src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector512.cs

```diff
@@ -469,8 +469,8 @@ public static void CopyTo<T>(this Vector512<T> vector, T[] destination, int star
                 ThrowHelper.ThrowArgumentException_DestinationTooShort();
             }
 
-            ref byte address = ref Unsafe.As<T, byte>(ref MemoryMarshal.GetArrayDataReference(destination));
-            Unsafe.WriteUnaligned(ref Unsafe.Add(ref address, startIndex), vector);
+            ref T address = ref MemoryMarshal.GetArrayDataReference(destination);
```

Comment:
Rather than changing 7 other files, you should probably just make this match what `V256` was doing.

It is a smaller change, more consistent, and avoids introducing new locals.

A regression test likewise needs to be added